IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -92,7 +92,8 @@ - (BOOL)saveAccessToken:(MSIDToken *)token
     
     for (MSIDToken *tokenInCache in allTokens)
     {
-        if ([tokenInCache.authority msidIsEquivalentAuthority:token.authority]
+        NSArray<NSURL *> *aliases = [[MSIDAadAuthorityCache sharedInstance] cacheAliasesForAuthority:tokenInCache.authority];
+        if ([tokenInCache.authority msidIsEquivalentWithAnyAlias:aliases]
```

Comment:
Discussed offline that this logic needs to be optimized, because underlying method does additional filtering as well